web/core/src/jsTest/kotlin/elements/EventTests.kt

```diff
@@ -36,6 +39,165 @@ class EventTests {
         assertTrue(handeled)
     }
 
+    @Test
+    fun multipleEventListenersCanBeAddedAndUsed() = runTest {
+        var handled1 = false
+        var handled2 = false
+
+        composition {
+            Button(
+                {
+                    onClick { handled1 = true }
+                    onClick { handled2 = true }
+                }
+            ) {}
+        }
+
+        assertEquals(1, root.childElementCount)
+
+        val btn = root.firstChild as HTMLElement
+        btn.dispatchEvent(MouseEvent("click"))
+
+        assertTrue(handled1)
+        assertTrue(handled2)
+    }
+
+    @Test
+    fun multipleButtonClickHandled() = runTest {
+        var count = 0
+
+        composition {
+            Button(attrs = { onClick { count++ } })
+        }
+
+        assertEquals(1, root.childElementCount)
+
+        val btn = root.firstChild as HTMLElement
+        btn.dispatchEvent(MouseEvent("click"))
+        btn.dispatchEvent(MouseEvent("click"))
+        btn.dispatchEvent(MouseEvent("click"))
+
+        assertEquals(3, count)
+    }
+
+    @Test
+    fun buttonOnClickHandlerWithOnceOptionCalledOnlyOnce() = runTest {
+        var count = 0
+
+        composition {
+            Button(attrs = { onClick(options = Options(once = Options.BooleanValue.True)) { count++ } })
+        }
+
+        assertEquals(1, root.childElementCount)
+
+        val btn = root.firstChild as HTMLElement
+        btn.dispatchEvent(MouseEvent("click"))
+        btn.dispatchEvent(MouseEvent("click"))
+        btn.dispatchEvent(MouseEvent("click"))
+
+        assertEquals(1, count)
+    }
+
+    @Test
+    fun buttonOnClickHandlerWithOnceOptionCalledAgainAfterRecomposition() = runTest {
+        var count = 0
+        var composedTimes = 0
+
+        var recomposeScope: RecomposeScope? = null
+
+        composition {
+            recomposeScope = currentRecomposeScope
+            composedTimes++
+
+            Button(attrs = { onClick(options = Options(once = Options.BooleanValue.True)) { count++ } })
```

Comment:
From what I see this PR is about two things, not one - we are introducing syntax for one-time events for instance. I don't think it's safe enough to shove in two different big changes in one PR.  But let's discuss

Reply:
that's right.
the `Options` changes are now not a part of this PR